### PR TITLE
nrf cloud REST API sample

### PR DESCRIPTION
**Pull request by jayteemo**

### CODE REVIEW

File: samples/spm/prj_minimal.conf.txt [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#
```
Comment (plskeggs) [inline samples/spm/prj_minimal.conf.txt]: why was this file changed?

File: subsys/spm/Kconfig [hunk 2]
```diff
@@ -25,12 +25,16 @@ menuconfig IS_SPM
 	select TRUSTED_EXECUTION_SECURE
 
 if IS_SPM
+
```
Comment (plskeggs) [inline subsys/spm/Kconfig]: Why was this changed?  Part of a PR you cherry picked for local testing?

Oh, yeah... this thing: https://github.com/nrfconnect/sdk-nrf/pull/4103.

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_rest.c [hunk 3]
```diff
@@ -0,0 +1,647 @@
...
+#include "nrf_cloud_codec.h"
+
+LOG_MODULE_REGISTER(nrf_cloud_rest, CONFIG_NRF_CLOUD_REST_LOG_LEVEL);
+
+#define HTTPS_PORT 443
+#define HOSTNAME "api.nrfcloud.com"
+#define HOSTNAME_DEV "api.dev.nrfcloud.com"
+#define TLS_SEC_TAG 16842753 //42
+#define BASE64_PAD_CHAR '='
+#define GET_BASE64_LEN(n) (((4 * n / 3) + 3) & ~3)
```
Comment (plskeggs) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud_rest.c]: (4 * n / 3) should be clarified with more parens: ((4 *n) / 3)

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_rest.c [hunk 4]
```diff
@@ -0,0 +1,647 @@
...
+		return err;
+	}
+
+	err = setsockopt(fd, SOL_SOCKET, SO_RCVTIMEO,
+			 &timeout, sizeof(timeout));
+	if (err) {
+		LOG_ERR("Failed to set socket recv timeout, error: %d/n", errno);
+		return err;
+	}
+
```
Comment (plskeggs) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud_rest.c:236]: Nice to see these timeouts being used.  Have you seen them actually time out?

File: samples/nrf9160/nrf_cloud_rest_api/src/main.c [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+		LOG_INF("No pending FOTA jobs");
+		goto cleanup;
+	}
+
+	LOG_INF("FOTA Job: %s, type: %d", log_strdup(job.id), job.type);
+
+	/* Disconnect after next API call */
+	rest_ctx.keep_alive = false;
+
+	/* Mark the job as succeeded */
```
Comment (plskeggs) [inline samples/nrf9160/nrf_cloud_rest_api/src/main.c]: Does this actually do the FOTA update somewhere else, or is that left as an exercise for the reader?

File: subsys/spm/Kconfig (hunk 2 above)
Comment (jayteemo) [reply subsys/spm/Kconfig]: Yeah this and the other spm change are from the pr that fixed the broken builds for thingys.  You weren’t suppose to look at these 😛

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_rest.c (hunk 4 above)
Comment (jayteemo) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_rest.c:236]: I have not

File: samples/nrf9160/nrf_cloud_rest_api/src/main.c (hunk 5 above)
Comment (jayteemo) [reply samples/nrf9160/nrf_cloud_rest_api/src/main.c]: Yeah this is just to exercise the api.

Comment (plskeggs) [reply samples/nrf9160/nrf_cloud_rest_api/src/main.c]: You might leave a comment to that effect, in case someone starts with this to develop something real.
